File: packages/sdk-tests/js/tests/health/health.test.ts
import { describe, it, expect } from "vitest";

const BASE_URL = process.env.HIVE_BASE_URL ?? "http://localhost:8080/v1";

/** Derive the root URL by stripping the /v1 suffix. */
function rootURL(): string {
  return BASE_URL.replace(/\/v1\/?$/, "");
}

describe("Health endpoint", () => {
  it("returns 200 with status ok", async () => {
    const res = await fetch(`${rootURL()}/health`);

    expect(res.status).toBe(200);

    const body = await res.json();
    expect(body).toEqual({ status: "ok" });
  });
});

File: packages/sdk-tests/java/src/test/java/com/hive/sdktests/HealthTest.java
package com.hive.sdktests;

import static org.junit.jupiter.api.Assertions.*;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import org.junit.jupiter.api.Test;

class HealthTest {

    private static final String BASE_URL =
            System.getenv("HIVE_BASE_URL") != null
                    ? System.getenv("HIVE_BASE_URL")
                    : "http://localhost:8080/v1";

    /** Derive root URL by stripping the /v1 suffix. */
    private static String rootURL() {
        return BASE_URL.replaceAll("/v1/?$", "");
    }

    @Test
    void healthEndpointReturns200WithStatusOk() throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request =
                HttpRequest.newBuilder()
                        .uri(URI.create(rootURL() + "/health"))
                        .GET()
                        .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        assertEquals(200, response.statusCode());
        assertTrue(response.body().contains("\"status\""));
        assertTrue(response.body().contains("\"ok\""));
    }
}
